core/src/Streamly/Internal/Data/Stream/Expand.hs

```diff
@@ -584,6 +584,10 @@ concatSmapMWith combine f initial =
 -- >>> Stream.fold Fold.toList $ Stream.concatPairsWith combine generate s
 -- [1,2,5,7,9]
 --
+-- Note that if the stream length is not a power of 2, the binary tree composed
+-- by concatPairsWith would not be balanced, which may or may not be important
+-- depending on what you are trying to achieve.
+--
```

Comment:
Is the `caution` below correct?
`the stream of streams must be finite`
In the example above, the stream of streams is not infinite.

Reply:
Not sure I understand. it is consistent, the example is finite and caution is also finite.

Reply:
Ah, my bad. I read finite as `infinite`.